src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -85,35 +86,89 @@ public enum ModuleCreationOption {
 
     /**
-     * Creates {@link DefaultConfiguration} instance for the given module class.
+     * Returns test logger.
      *
-     * @param clazz module class.
-     * @return {@link DefaultConfiguration} instance.
+     * @return logger for tests
      */
-    protected abstract DefaultConfiguration createModuleConfig(Class<?> clazz);
+    protected final BriefUtLogger getBriefUtLogger() {
+        return new BriefUtLogger(stream);
+    }
 
     /**
-     * Returns test logger.
+     * Creates a default module configuration {@link DefaultConfiguration} for a given object
+     * of type {@link Class}.
      *
-     * @return logger test logger
+     * @param clazz a {@link Class} type object.
+     * @return default module configuration for the given {@link Class} instance.
      */
-    protected final BriefUtLogger getBriefUtLogger() {
-        return new BriefUtLogger(stream);
+    protected static DefaultConfiguration createModuleConfig(Class<?> clazz) {
+        return new DefaultConfiguration(clazz.getName());
     }
 
     /**
-     * Returns canonical path for the file with the given file name.
-     * The path is formed base on the non-compilable resources location.
-     * This implementation uses 'src/test/resources-noncompilable/'
-     * as a non-compilable resource location.
+     * Returns {@link Configuration} instance for the given module name pulled
+     * from the {@code masterConfig}.
      *
-     * @param filename file name.
-     * @return canonical path for the file with the given file name.
-     * @throws IOException if I/O exception occurs while forming the path.
+     * @param masterConfig The master configuration to examine.
+     * @param moduleName module name.
+     * @param moduleId module id.
+     * @return {@link Configuration} instance for the given module name.
+     * @throws IllegalStateException if there is a problem retrieving the module
+     *         or config.
      */
-    protected final String getNonCompilablePath(String filename) throws IOException {
-        return new File("src/" + getResourceLocation() + "/resources-noncompilable/"
-                + getPackageLocation() + "/" + filename).getCanonicalPath();
+    protected static Configuration getModuleConfig(Configuration masterConfig, String moduleName,
+            String moduleId) {
+        final Configuration result;
+        final List<Configuration> configs = getModuleConfigs(masterConfig, moduleName);
+        if (configs.size() == 1) {
+            result = configs.get(0);
+        }
+        else if (configs.isEmpty()) {
+            throw new IllegalStateException("no instances of the Module was found: " + moduleName);
+        }
+        else if (moduleId == null) {
+            throw new IllegalStateException("multiple instances of the same Module are detected");
+        }
+        else {
+            result = configs.stream().filter(conf -> {
+                try {
```

Comment:
Please extract this to a helper method.
try/catch lambdas looks weird.

Reply:
Done.